### PR TITLE
doc dir with tmp reference file + docker files into own dir

### PR DESCRIPTION
**Pull request by RogueM**

### CODE REVIEW

**Comment by alexellis** [comment]
So, normally a Dockerfile is in the root of a project, I've tested it in its previous location but not sure what impact your change will have. Have you fully tested the changes (relative paths etc are involved)? @RogueM

**Comment by RogueM** [comment]
no, I was talking to Phil about it... the change *should* have no impact, since your scripts essentially copy what's necessary to the parent and run from there, but it's untested.

I just want to avoid files in the software tree in preparation for debian building (it shouldn't matter but it's still good to not have unrelated files there IMO).

**Comment by alexellis** [comment]
It was placed tentatively in the sub-folder to make it easier for @Gadgetoid to accept into the repo. A `Dockerfile` will generally live in the root of the project so it has access to the project at the top level. It will also give visibility to people who know Docker when they work with a repo they know immediately.

> You can also use Docker to build the deb file... which is a very cool thing for another discussion.

Had a quick look and there appear to be some easy ways to ignore files from debian packages:

* http://serverfault.com/questions/247120/how-can-i-ignore-files-directories-when-building-debian-packages
* https://www.debian.org/doc/manuals/maint-guide/dother.en.html#compat

If you guys are really keen to move the files then it is worth giving the build a quick test first. Instructions for installing Docker on Raspbian at: [Quick Start on alexellis/docker-arm](https://github.com/alexellis/docker-arm/blob/master/ZERO.md)

File: docker/DOCKER.md [hunk 1]
```diff
@@ -4,7 +4,6 @@ Fully tested Dockerfile for using scroll-phat library without needing to install
```
Comment (alexellis) [inline docker/DOCKER.md]: May make sense to promote these instructions to top level of repo?

**Comment by RogueM** [comment]
yes, it's totally possible yo work around if necessary, but we're trying to structure all repos similarly to avoid repo specific issues or customisation for software distribution purposes.

... by root of the project, do you mean inside the software tree (as it was)? since your script copies the DockerFile there and executes from there then my assumption was that whether they sit in the 'library' or a 'docker' dir placed at the same level?

I guess, yes, I can confirm that's the case, but I'm concurrently working on about 12 repos... the best way is if you can confirm the change is fine yoursellf, perhaps using the testing branch of my fork. Otherwise then I'll scrap the PR and see in a few weeks time what's sane and not sane to do.

**Pull request closed without merging**